src/Services/ComposerConflictsRenovator.php

```diff
@@ -0,0 +1,258 @@
+                            $software_name,
+                            $cvss,
+                            $updated_result->getConflictVersionsString() ?? '',
+                            implode(' , ', $software[0]['references'] ?? [])
+                        )
+                    );
+                } catch (Exception $e) {
+                    continue;
+                }
+            }
+
+            sleep($pause);
+        }
+    }
+
+    /**
+     * Updates composer.json conflict section for specified vulnerability.
+     *
+     * @param array $entry
+     * @param array $composer_json_content
+     *
+     * @return ConflictSectionUpdateResult|null
+     */
+    protected function updateConflictsForVulnerability(
+        array $entry,
+        array $composer_json_content
+    ): ?ConflictSectionUpdateResult {
+        $updated  = false;
+        $software = $entry['software'];
+        if (empty($software)) {
+            return null;
+        }
+
+        foreach ($software as $software_entry) {
+            $affectedVersions = $software_entry['affected_versions'] ?? [];
+            if (!is_array($affectedVersions) || empty($affectedVersions)) {
+                continue;
+            }
+
+            $entry_type = $software_entry['type'];
+            switch ($entry_type) {
+                case 'plugin':
+                case 'theme':
+                    $vulnerability_key =
+                        strtolower(sprintf('wpackagist-%1$s/%2$s', $entry_type, $software_entry['slug']));
+
+                    break;
+
+                case 'core':
+                    if ($software_entry['slug'] !== 'wordpress') {
+                        break;
+                    }
+
+                    $vulnerability_key = 'roots/wordpress';
+
+                    break;
+
+                default:
+                    break;
+            }
+
+            if (!isset($vulnerability_key)) {
+                continue;
+            }
+
+            foreach ($affectedVersions as $affected_version) {
+                $conflict_versions_string = $this->getConflictStringForAffectedVersion($affected_version);
+                if (!isset($conflict_versions_string)) {
+                    continue;
+                }
+
+                if (!isset($composer_json_content['conflict'][$vulnerability_key])) {
+                    $composer_json_content['conflict'][$vulnerability_key] = $conflict_versions_string;
+                    $updated                                               = true;
+
+                    continue;
+                }
+
+                if (!str_contains($composer_json_content['conflict'][$vulnerability_key], $conflict_versions_string)) {
+                    $composer_json_content['conflict'][$vulnerability_key] .= " || {$conflict_versions_string}";
+                    $updated                                               = true;
+                }
+            }
+
+            if ($updated) {
```

Comment:
it looks like there is some issue with the `$updated` variable, because it's only set to `false` at the start of the method, then once it's set to true in the second foreach, it will keep calling `ksort`

Reply:
fixed